HighAvailabilityModule.Interface/IMembershipStorageClient.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace HighAvailabilityModule.Interface
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    public interface IMembershipStorageClient
+    {
+        TimeSpan OperationTimeout { get; set; }
+
+        Task<(string, string)> GetDataEntryAsync(string path, string key);
+        Task<Guid> TryGetGuid(string path, string key);
+        Task<string> TryGetString(string path, string key);
+        Task<int> TryGetInt(string path, string key);
+        Task<long> TryGetLong(string path, string key);
+        Task<double> TryGetDouble(string path, string key);
+        Task<string[]> TryGetStringArray(string path, string key);
+        Task<byte[]> TryGetByteArray(string path, string key);
+
+        Task SetDataEntryAsync(string path, string key, string value, string type);
```

Comment:
This method is not safe for every possible string `type`. To avoid hand made type checker, let's hide this method from our interface.